### PR TITLE
chore: update Concrete Python to 2.5.1

### PR DESCRIPTION
**Pull request by fd0r**

### CODE REVIEW

**Review by andrei-stoian-zama — APPROVED** [review]

File: src/concrete/ml/sklearn/qnn.py [hunk 1]
```diff
@@ -141,7 +141,8 @@ def __init__(
         virtual_params = kwargs.pop("virtual_params_", {})
 
         self._kwargs = kwargs
-        vars(self).update(kwargs)
```
Comment (kcelia) [inline src/concrete/ml/sklearn/qnn.py:144]: Out of curiosity, why did you change it ? 
It's the same, and the first one is more concise.

Comment (fd0r) [reply src/concrete/ml/sklearn/qnn.py:144]: I had to update mypy that wasn't pleased with this since `vars` doesn't return a `dict` but a `mapping` that in the spec doesn't have an `update` method.
